adapters/smaato/smaato.go

```diff
@@ -298,22 +432,85 @@ func parseImpressionObject(imp *openrtb2.Imp) error {
 		return nil
 	}
 
-	return fmt.Errorf("invalid MediaType. SMAATO only supports Banner and Video. Ignoring ImpID=%s", imp.ID)
+	return &errortypes.BadInput{Message: "Invalid MediaType. Smaato only supports Banner and Video."}
 }
 
-func extractUserExtAttributes(userExt userExt, userCopy *openrtb2.User) {
-	gender := userExt.Data.Gender
-	if gender != "" {
-		userCopy.Gender = gender
+func setImpForAdBreak(imps []openrtb2.Imp) error {
+	adBreakID, err := jsonparser.GetString(imps[0].Ext, "bidder", "adbreakId")
```

Comment:
Please consider adding a check to make sure the length of imps is greater than 0 before accessing the 0th imp

Reply:
it shouldn't be possible that imps is empty, but nevertheless i added a check